src/passes/Asyncify.cpp

```diff
@@ -1266,22 +1289,37 @@ struct Asyncify : public Pass {
     String::Split listedImports(stateChangingImports, ",");
     auto ignoreIndirect = runner->options.getArgumentOrDefault(
                             "asyncify-ignore-indirect", "") == "";
-    String::Split blacklist(
-      String::trim(read_possible_response_file(
-        runner->options.getArgumentOrDefault("asyncify-blacklist", ""))),
-      ",");
-    String::Split whitelist(
+    std::string removeListInput =
+      runner->options.getArgumentOrDefault("asyncify-removelist", "");
+    if (removeListInput.empty()) {
+      // Support old name for now to avoid immediate breakage TODO remove
+      removeListInput =
+        runner->options.getArgumentOrDefault("asyncify-blacklist", "");
+    }
+    String::Split removeList(
+      String::trim(read_possible_response_file(removeListInput)), ",");
+    String::Split addList(
       String::trim(read_possible_response_file(
-        runner->options.getArgumentOrDefault("asyncify-whitelist", ""))),
+        runner->options.getArgumentOrDefault("asyncify-addlist", ""))),
       ",");
+    std::string onlyListInput =
+      runner->options.getArgumentOrDefault("asyncify-onlylist", "");
+    if (onlyListInput.empty()) {
+      // Support old name for now to avoid immediate breakage TODO remove
+      onlyListInput =
+        runner->options.getArgumentOrDefault("asyncify-whitelist", "");
+    }
+    String::Split onlyList(
+      String::trim(read_possible_response_file(onlyListInput)), ",");
     auto asserts =
       runner->options.getArgumentOrDefault("asyncify-asserts", "") != "";
 
-    blacklist = handleBracketingOperators(blacklist);
-    whitelist = handleBracketingOperators(whitelist);
+    removeList = handleBracketingOperators(removeList);
+    addList = handleBracketingOperators(addList);
+    onlyList = handleBracketingOperators(onlyList);
 
-    if (!blacklist.empty() && !whitelist.empty()) {
-      Fatal() << "It makes no sense to use both a blacklist and a whitelist "
+    if (!removeList.empty() && !onlyList.empty()) {
+      Fatal() << "It makes no sense to use both a remove list and an only list "
```

Comment:
Should we add a similar check for add list + only list? Otherwise the documentation is ambiguous about whether the add list items will get added to the only list items.

Reply:
Good idea, added. Yeah, the only-list doesn't really make sense together with anything else.